csharp_dotnetcore/14.NLP-With-Dispatch/Startup.cs

```diff
@@ -0,0 +1,255 @@
+                options.Middleware.Add(appInsightsLogger);
+            });
+        }
+
+        /// <summary>
+        /// This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        /// </summary>
+        /// <param name="app">The application builder.  This provides the mechanisms to configure an application's request pipeline.</param>
+        /// <param name="env">Provides information about the web hosting environment an application is running in.</param>
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            app.UseDefaultFiles()
+                .UseStaticFiles()
+                .UseBotFramework();
+        }
+
+        /// <summary>
+        /// Initialize the bot's references to external services.
+        ///
+        /// For example, Application Insights and QnaMaker services
+        /// are created here.  These external services are configured
+        /// using the <see cref="BotConfiguration"/> class (based on the contents of your ".bot" file).
+        /// </summary>
+        /// <param name="config">Configuration object based on your ".bot" file.</param>
+        /// <returns>A <see cref="BotConfiguration"/> representing client objects to access external services the bot uses.</returns>
+        /// <seealso cref="BotConfiguration"/>
+        /// <seealso cref="QnAMaker"/>
+        /// <seealso cref="TelemetryClient"/>
+        private static BotServices InitBotServices(BotConfiguration config)
+        {
+            TelemetryClient telemetryClient = null;
+            var qnaServices = new Dictionary<string, QnAMaker>();
+            var luisServices = new Dictionary<string, LuisRecognizer>();
+
+            foreach (var service in config.Services)
+            {
+                switch (service.Type)
+                {
+                    case ServiceTypes.Luis:
+                        {
+                            // Create a Luis Recognizer that is initialized and suitable for passing
+                            // into the IBot-derived class (NlpDispatchBot).
+                            // In this case, we're creating a custom class (wrapping the original
+                            // Luis Recognizer client) that logs the results of Luis Recognizer results
+                            // into Application Insights for future anaysis.
+                            var luis = (LuisService)service;
+                            if (luis == null)
+                            {
+                                throw new InvalidOperationException("The Luis service is not configured correctly in your '.bot' file.");
+                            }
+
+                            if (string.IsNullOrWhiteSpace(luis.AppId))
+                            {
+                                throw new InvalidOperationException("The Luis Model Application Id ('appId') is required to run this sample.  Please update your '.bot' file.");
```

Comment:
>   [](start = 140, length = 2)

Why 2 spaces on each of these after the "."?

Reply:
Not intended, it was a single mistake propagated by copy-pasting and then editing :)